internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubClientsFactory.java

```diff
@@ -23,47 +23,52 @@
  *
  * This will be passed as one of the arguments to the component which depends on the pub-sub APIs.
  */
-public class PubSubClientsFactory {
+public class PubSubClientsFactory<P extends PubSubProducerAdapter, C extends PubSubConsumerAdapter, A extends PubSubAdminAdapter> {
```

Comment:
Let’s make sure we update all usages to ensure type safety, if we care about maintaining it.

Reply:
@arjun4084346 - Can we add Type on individual `PubSubProducerAdapterFactory` and not on `PubSubClientsFactory`?